scripts/solve_network.py

```diff
@@ -130,16 +194,146 @@ def add_co2_sequestration_limit(n, limit=200):
         limit = float(o[o.find("seq") + 3 :]) * 1e6
         break
 
-    n.add(
+    if config["foresight"] == "perfect":
+        periods = n.investment_periods
+        names = pd.Index([f"co2_sequestration_limit-{period}" for period in periods])
+    else:
+        periods = [np.nan]
+        names = pd.Index(["co2_sequestration_limit"])
+
+    n.madd(
         "GlobalConstraint",
-        "co2_sequestration_limit",
+        names,
         sense="<=",
         constant=limit,
         type="primary_energy",
         carrier_attribute="co2_absorptions",
+        investment_period=periods,
     )
 
 
+def add_carbon_constraint(n, snapshots):
```

Comment:
Could you shortly point out why this one (and the one below) is necessary? It feels like this part of code should rather go into pypsa itself, but not sure...

Reply:
Mhm, I think one of the main reasons I implemented it directly in PyPSA-Eur is that with this implementation it is solving faster because the constraint is just on the CO2 atmosphere store and not on all components which have emissions. But this is maybe rather PyPSA-Eur specific since I guess many people do not model carbon management.